### PR TITLE
[CURA-8060] Fix combine layers (vertically) for support as well.

### PR DESCRIPTION
**Pull request by rburema**
Layers for support wouldn't combine vertically where this was needed, like for PVA printing. This relies on the work done in jira ticket CURA-7756 where almost, but not quite, the same thing happens for infill. In fact, it relies on that PR for how the infill generation class now has promoted the wall generating part to a publicly callable function, as one of the problems of both this ticket and its precursor was that (extra) walls _shouldn't_ combine. (And the 'normal' lines of course _should_ that's the whole point.)

See also:  #1426  --- please merge that one first!

### CODE REVIEW

File: src/FffGcodeWriter.cpp
Comment (jellespijker) [inline src/FffGcodeWriter.cpp:2500]: I would assign the pre-generated wall_toolpaths to this variable. Since the `generate()` still makes some use of them.
so in effect `VariableWidthPaths wall_toolpaths = part.wall_toolpaths`

File: src/FffGcodeWriter.cpp
Comment (jellespijker) [inline src/FffGcodeWriter.cpp:2526]: `constexpr` the wall line count of 0

File: src/FffGcodeWriter.cpp
Comment (jellespijker) [inline src/FffGcodeWriter.cpp:2523]: Dubbel check if the support infill overlap is correct. This infill overlap is also used when generating the wall toolpaths and the contour for the lines (used here) is generated from that "offset" inner_contour.

File: src/FffGcodeWriter.cpp
Comment (jellespijker) [inline src/FffGcodeWriter.cpp:2534]: If you assign the part.wall_toolpaths to the wall_toolpaths as suggested above. Don't forget to rollback these ;-)

**Review by jellespijker — CHANGES_REQUESTED** [review]

File: src/FffGcodeWriter.cpp
Comment (rburema) [reply src/FffGcodeWriter.cpp:2500]: Yeah I wanted to ask about that ... why a copy and not a reference?

File: src/FffGcodeWriter.cpp
Comment (rburema) [reply src/FffGcodeWriter.cpp:2526]: How is that different?

File: src/FffGcodeWriter.cpp
Comment (rburema) [reply src/FffGcodeWriter.cpp:2534]: If I do that, I won't forget ;-)
... I hope.

File: src/FffGcodeWriter.cpp
Comment (jellespijker) [reply src/FffGcodeWriter.cpp:2500]: I think it I did that because this function was const, and the generate could modify the wall tool paths.
Might be worth looking into if we can use reference

File: src/FffGcodeWriter.cpp
Comment (jellespijker) [reply src/FffGcodeWriter.cpp:2526]: Readability there are a lot of arguments in the function generate. Most IDE display the argument names but not all.
Can't hurt

File: src/FffGcodeWriter.cpp
Comment (rburema) [reply src/FffGcodeWriter.cpp:2500]: What function is const? part.(infill_)wall_toolpaths isn't? combineSupportInfillLayers isn't?

File: src/FffGcodeWriter.cpp
Comment (jellespijker) [reply src/FffGcodeWriter.cpp:2534]: I thought so I have high regards for your mental capabilities

File: src/FffGcodeWriter.cpp
Comment (jellespijker) [reply src/FffGcodeWriter.cpp:2500]: No but the FFGodewriter processSingleLayerInfill and processMultiLayerInfill is.

Comment (jellespijker) [reply src/FffGcodeWriter.cpp:2500]: quick call?

File: src/FffGcodeWriter.cpp
Comment (rburema) [reply src/FffGcodeWriter.cpp]: Haha, already decided to fix it as 'readability' after all before reading this comment :-)

File: src/FffGcodeWriter.cpp
Comment (rburema) [reply src/FffGcodeWriter.cpp]: Turns out I can't do it either :-p

File: src/FffGcodeWriter.cpp
Comment (rburema) [reply src/FffGcodeWriter.cpp]: Turns out I didn't forget.

**Review by jellespijker — APPROVED** [review]
Yet again you managed to get my stamp of approval. Once @nallath gives his approval to the changes made in PR #1426 I will merge this!

**Pull request merged**
